src/appleseed/renderer/kernel/lighting/directlightingintegrator.cpp

```diff
@@ -698,57 +676,46 @@ void DirectLightingIntegrator::add_non_physical_light_sample_contribution(
     light->evaluate(
         m_shading_context,
         sample.m_light_transform,
-        m_point,
+        m_material_sampler.get_point(),
         emission_position,
         emission_direction,
         light_value);
 
     // Compute the incoming direction in world space.
     const Vector3d incoming = -emission_direction;
 
-    // Cull light samples behind the shading surface if the BSDF is either reflective or transmissive,
-    // but not both.
-    if (m_bsdf.get_type() != BSDF::AllBSDFTypes)
+    if (m_material_sampler.cull_incoming_direction(incoming))
     {
```

Comment:
I suggest omitting the braces for single-line bodies:

    if (m_material_sampler.cull_incoming_direction(incoming))
        return;